core/types/system_tx.go

```diff
@@ -0,0 +1,87 @@
+package types
+
+import (
+	"bytes"
+	"math/big"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/rlp"
+)
+
+type SystemTx struct {
+	From  common.Address  // pre-determined sender
```

Comment:
What `From` used for? For access control in the contract?

Reply:
yeah, i think at least. I'll check with haichen, maybe we dont need it at all.

Reply:
yeah we need it, otherwise arbitrary signers could send txs to the system contract that can modify its state.